neo/rawio/plexon2rawio/plexon2rawio.py

```diff
@@ -0,0 +1,499 @@
+
+Note on IO performance and memory consumption:
+This IO is only partially lazy: To load a data chunk of an individual
+signal channels, the complete data of that channel will be loaded into memory
+and cached. This occurs as a limitation of the pypl2 package and might be improved
+in future versions by exploiting the block-wise loading of signals.
+
+To clear the currently cached signal data use the `clear_analogsignal_cache()` method.
+
+Note on Neo header and spike channels:
+The IO only considers enabled channels and will not list disabled channels in its header.
+
+There is no 1-1 correspondence of PL2 spike channels and neo spike channels as each unit of
+a PL2 spike channel will be represented as an individual neo spike channel.
+
+Author: Julia Sprenger
+"""
+import pathlib
+import warnings
+from collections import namedtuple
+from urllib.request import urlopen
+from datetime import datetime
+
+import numpy as np
+
+
+from ..baserawio import (BaseRawIO, _signal_channel_dtype, _signal_stream_dtype,
+                         _spike_channel_dtype, _event_channel_dtype)
+
+
+class Plexon2RawIO(BaseRawIO):
+    """
+    Class for "reading" data from a PL2 file
+
+    Usage:
+        >>> import neo.rawio
+        >>> r = neo.rawio.Plexon2RawIO(filename='my_data.pl2')
+        >>> r.parse_header()
+        >>> print(r)
+        >>> raw_chunk = r.get_analogsignal_chunk(block_index=0, seg_index=0,
+                            i_start=0, i_stop=1024,  stream_index=0, channel_indexes=range(10))
+        >>> float_chunk = r.rescale_signal_raw_to_float(raw_chunk, dtype='float64',
+                            channel_indexes=[0, 3, 6])
+        >>> spike_timestamp = r.get_spike_timestamps(spike_channel_index=0, t_start=None,
+                            t_stop=None)
+        >>> spike_times = r.rescale_spike_timestamp(spike_timestamp, 'float64')
+        >>> ev_timestamps, _, ev_labels = r.get_event_timestamps(event_channel_index=0)
+
+    """
+    extensions = ['pl2']
+    rawmode = 'one-file'
+
+    def __init__(self, filename, pl2_dll_file_path=None):
+
+        # signals, event and spiking data will be cached
+        # cached signal data can be cleared using `clear_analogsignal_cache()()`
+        self._analogsignal_cache = {}
+        self._event_channel_cache = {}
+        self._spike_channel_cache = {}
+
+        # note that this filename is used in self._source_name
+        self.filename = pathlib.Path(filename)
+
+        if (not self.filename.exists()) or (not self.filename.is_file()):
+            raise ValueError(f'{self.filename} is not a file.')
+
+        BaseRawIO.__init__(self)
+
+        # download default PL2 dll once if not yet available
+        if pl2_dll_file_path is None:
+            pl2_dll_folder = pathlib.Path .home() / '.plexon_dlls_for_neo'
+            pl2_dll_folder.mkdir(exist_ok=True)
+            pl2_dll_file_path = pl2_dll_folder / 'PL2FileReader.dll'
```

Comment:
Hi @JuliaSprenger 
This pull request was super helpful!!
Just one thing, while executing Plexon2RawIO(). I got the error:
OSError: [WinError 193] %1 is not a valid Win32 application
and
OSError: Error: Can't load PL2FileReader.dll at: C:\Users\admin.plexon_dlls_for_neo\PL2FileReader.dll. PL2FileReader.dll is bundled with the C++ PL2 Offline Files SDK located on the Plexon Inc website: www.plexon.com Contact Plexon Support for more information: support@plexon.com

I then realized my pypl2 SDK folder had another dll file called "PL2FileReader64.dll"
so after replace here "PL2FileReader.dll" with "PL2FileReader64.dll", it started running smoothly.

Just wanted to keep a note here if anyone else encounters those errors and wants to move ahead